File: SagaPlay-Backend/UserService/Models/UserProfile.cs
using Microsoft.EntityFrameworkCore;

namespace UserService.Models
{
    public class UserProfile
    {
        public Guid UserProfileId { get; set; }

        //Foreign Key to Users
        public Guid UserId { get; set; }

        //Navigation property
        public User User { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }      

        public string EmailAddress { get; set; }

        public DateTime DateofBirth { get; set; }

        public string Bio { get; set; }

        public string ProfilePictureUrl { get; set; }

        public string Country { get; set; }

        public string PhoneNumber { get; set; }


    }
}

File: SagaPlay-Frontend/src/app/features/user-profile/user-profile.ts
export interface User {
    Profile: UserProfile;
    Preferences: UserPreferences;
}

export interface UserProfile {
    UserId: string;    
    FirstName: string;
    LastName: string;
    EmailAddress: string;
    DateofBirth: string;
    Bio: string;
    ProfilePictureUrl: string;
    Country: string;
    PhoneNumber: string;
}

export interface UserPreferences {
    UserId: string;
    Theme: string;
    Language: string;
    NotificationSettings: string;
    PlayBackQualitySettings: string;
    ReceiveNewsLetter: boolean;
}